cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```diff
@@ -76,15 +76,19 @@ public void setRepository(FullHttpRequest request, HttpResponder responder,
       }
 
       repoRequest.getConfig().validate();
+      
       if (repoRequest.shouldTest()) {
-        // TODO: CDAP-20252, add the validate logic once the SourceControlManager module is ready
+        sourceControlService.testRepositoryConfig(namespace, repoRequest.getConfig());
+        responder.sendJson(HttpResponseStatus.OK,
+                           GSON.toJson(
+                             new SetRepositoryResponse("Successfully validated the repository configuration.")));
+      } else {
+        sourceControlService.setRepository(namespace, repoRequest.getConfig());
+        responder.sendJson(HttpResponseStatus.OK,
```

Comment:
looks like you are only changing the response message. Maybe can move the responder call outsider and pass the message variable

Reply:
nice sugguestion!